typescript/infra/src/core/govern.ts

```diff
@@ -55,86 +147,66 @@ export class AbacusCoreGovernor<Chain extends ChainName> {
-        await mapFn(connection, calls, chain as Chain);
+  protected async inferCallSubmissionTypes() {
+    for (const chain of Object.keys(this.calls) as Chain[]) {
+      for (const call of this.calls[chain]) {
+        const submissionType = await this.inferCallSubmissionType(chain, call);
+        call.submissionType = submissionType;
       }
     }
   }
 
-  connectionFn = (chain: Chain) => {
-    return this.checker.multiProvider.getChainConnection(chain);
-  };
-
-  /*
-  // NB: Add this back in order to run using a Ledger signer.
-  import { LedgerSigner } from '@ethersproject/hardware-wallets';
-
-  // Due to TS funkiness, this needs to be imported in order for this
-  // code to build, but needs to be removed in order for the code to run.
-  import '@ethersproject/hardware-wallets/thirdparty';
-
-  ledgerConnectionFn = (chain: Chain) => {
+  protected async inferCallSubmissionType(
+    chain: Chain,
+    call: AnnotatedCallData,
+  ): Promise<SubmissionType> {
     const connection = this.checker.multiProvider.getChainConnection(chain);
-    // Ledger Live derivation path, vary the third number  to select different
-    // accounts.
-    const path = "m/44'/60'/2'/0/0";
-    return new ChainConnection({
-      signer: new LedgerSigner(connection.provider, 'hid', path),
-      provider: connection.provider,
-      overrides: connection.overrides,
-      confirmations: connection.confirmations,
-    });
-  };
-  */
+    // 1. Check if the call will succeed with the default signer.
+    try {
+      await connection.estimateGas(call);
+      return SubmissionType.SIGNER;
+    } catch (_) {} // eslint-disable-line no-empty
 
-  protected async estimateFn(
-    connection: ChainConnection,
-    calls: types.CallData[],
-  ) {
-    await Promise.all(calls.map((call) => connection.estimateGas(call)));
-  }
+    try {
+      // 2. Check if the call will succeed via Gnosis Safe.
 
-  protected async sendFn(connection: ChainConnection, calls: types.CallData[]) {
-    for (const call of calls) {
-      await connection.sendTransaction(call);
-    }
-  }
+      // 2a. Check if calling from the owner will succeed.
+      const safeAddress = this.checker.configMap[chain!].owner;
+      if (!safeAddress) throw new Error(`Safe address not found for ${chain}`);
+      await connection.provider.estimateGas({
+        ...call,
+        from: safeAddress,
+      });
 
-  estimateCalls() {
-    return this.mapCalls(this.connectionFn, this.estimateFn);
-  }
+      // 2b. Confirm that the signer is a Safe owner or delegate.
+      // This should implicitly check whether or not the owner is a gnosis
+      // safe.
+      const signer = connection.signer;
+      if (!signer) throw new Error(`no signer found`);
+      const signerAddress = await signer.getAddress();
+      const proposer = await canProposeSafeTransactions(
+        signerAddress,
+        chain,
+        connection,
+        safeAddress,
+      );
+      if (!proposer)
```

Comment:
what about if (proposer) return safe instead of this throws
and the try catch only around estimate gas?